OpenRA.Mods.Common/Traits/ProductionParadrop.cs

```diff
@@ -0,0 +1,170 @@
+
+							foreach (var cargo in self.TraitsImplementing<INotifyDelivery>())
+								cargo.Delivered(self);
+							
+							self.World.AddFrameEndTask(ww => DoProduction(self, producee, exit, factionVariant));
+							Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", info.ReadyAudio, self.Owner.Faction.InternalName);
+						}));
+
+					actor.QueueActivity(new Fly(actor, Target.FromCell(w, endPos)));
+					actor.QueueActivity(new RemoveSelf());
+				});
+
+			return true;
+		}
+
+		public override void DoProduction(Actor self, ActorInfo producee, ExitInfo exitinfo, string factionVariant)
+		{
+			var exit = CPos.Zero;
+			var exitLocation = CPos.Zero;
+			var target = Target.Invalid;
+
+			var info = (ProductionParadropInfo)Info;
+			var actorType = info.ActorType;
+
+			var bi = producee.TraitInfoOrDefault<BuildableInfo>();
+			if (bi != null && bi.ForceFaction != null)
+				factionVariant = bi.ForceFaction;
+
+			var altitude = self.World.Map.Rules.Actors[actorType].TraitInfo<AircraftInfo>().CruiseAltitude;
+			var td = new TypeDictionary
+			{
+				new OwnerInit(self.Owner),
+			};
+
+			if (self.OccupiesSpace != null)
+			{
+				exit = self.Location + exitinfo.ExitCell;
+				var spawn = self.CenterPosition + exitinfo.SpawnOffset + new WVec(WDist.Zero, WDist.Zero, altitude);
+				var to = self.World.Map.CenterOfCell(exit);
+
+				var fi = producee.TraitInfoOrDefault<IFacingInfo>();
+				var initialFacing = exitinfo.Facing < 0 ? Util.GetFacing(to - spawn, fi == null ? 0 : fi.GetInitialFacing()) : exitinfo.Facing;
+
+				exitLocation = rp.Value != null ? rp.Value.Location : exit;
+				target = Target.FromCell(self.World, exitLocation);
+
+				td.Add(new LocationInit(exit));
+				td.Add(new CenterPositionInit(spawn));
+				td.Add(new FacingInit(initialFacing));
+			}
+
+			self.World.AddFrameEndTask(w =>
+				{
+					if (factionVariant != null)
+						td.Add(new FactionInit(factionVariant));
+
+					var newUnit = self.World.CreateActor(producee.Name, td);
+
+					newUnit.QueueActivity(new Parachute(newUnit,newUnit.CenterPosition + new WVec(WDist.Zero, WDist.Zero, altitude), self));
+
+					var move = newUnit.TraitOrDefault<IMove>();
+					if (move != null)
+					{
+						if (exitinfo.MoveIntoWorld)
+						{
+							if (exitinfo.ExitDelay > 0)
+								newUnit.QueueActivity(new Wait(exitinfo.ExitDelay, false));
+
+							newUnit.QueueActivity(move.MoveIntoWorld(newUnit, exit));
+							newUnit.QueueActivity(new AttackMoveActivity(
+								newUnit, move.MoveTo(exitLocation, 1)));
+						}
+					}
+
+					newUnit.SetTargetLine(target, rp.Value != null ? Color.Red : Color.Green, false);
+
+					if (!self.IsDead)
+						foreach (var t in self.TraitsImplementing<INotifyProduction>())
+							t.UnitProduced(self, newUnit, exit);
+
+					var notifyOthers = self.World.ActorsWithTrait<INotifyOtherProduction>();
+					foreach (var notify in notifyOthers)
+						notify.Trait.UnitProducedByOther(notify.Actor, self, newUnit);
```

Comment:
It isn't needed, and I don't think we have a use for it but that doesn't mean there can't be an Info flag regarding this and only notifying if that flag is true.

But if you do keep this code in any form you need to be sure you don't notify this production structure.
Look at `ClonesProducedUnits` for more information.